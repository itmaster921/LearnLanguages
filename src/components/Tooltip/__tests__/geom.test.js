import {
  Size,
  Rect,
  computeTopGeometry,
  computeBottomGeometry,
  computeLeftGeometry,
  computeRightGeometry,
} from '../src/geom';

const options1 = {
  displayArea: new Rect(0, 0, 375, 667),
  childRect: new Rect(24, 350, 64, 64),
  contentSize: new Size(200, 100),
  arrowSize: new Size(16, 8),
};

const options2 = {
  displayArea: new Rect(0, 0, 375, 667),
  childRect: new Rect(240, 350, 64, 64),
  contentSize: new Size(150, 200),
  arrowSize: new Size(7, 18),
};

const options3 = {
  displayArea: new Rect(0, 0, 375, 667),
  childRect: new Rect(24, 35, 150, 200),
  contentSize: new Size(300, 500),
  arrowSize: new Size(16, 8),
};

describe('Testing Computing Geometry', () => {
  it('correctly calculates top geometry', () => {
    expect(computeTopGeometry(options1)).toMatchSnapshot();
    expect(computeTopGeometry(options2)).toMatchSnapshot();
    expect(computeTopGeometry(options3)).toMatchSnapshot();
  });

  it('correctly calculates bottom geometry', () => {
    expect(computeBottomGeometry(options1)).toMatchSnapshot();
    expect(computeBottomGeometry(options2)).toMatchSnapshot();
    expect(computeBottomGeometry(options3)).toMatchSnapshot();
  });

  it('correctly calculates left geometry', () => {
    expect(computeLeftGeometry(options1)).toMatchSnapshot();
    expect(computeLeftGeometry(options2)).toMatchSnapshot();
    expect(computeLeftGeometry(options3)).toMatchSnapshot();
  });

  it('correctly calculates right geometry', () => {
    expect(computeRightGeometry(options1)).toMatchSnapshot();
    expect(computeRightGeometry(options2)).toMatchSnapshot();
    expect(computeRightGeometry(options3)).toMatchSnapshot();
  });
});
