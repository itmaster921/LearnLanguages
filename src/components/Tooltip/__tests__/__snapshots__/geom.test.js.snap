// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Computing Geometry correctly calculates bottom geometry 1`] = `
Object {
  "anchorPoint": Point {
    "x": 56,
    "y": 414,
  },
  "placement": "bottom",
  "tooltipOrigin": Point {
    "x": 0,
    "y": 422,
  },
}
`;

exports[`Testing Computing Geometry correctly calculates bottom geometry 2`] = `
Object {
  "anchorPoint": Point {
    "x": 272,
    "y": 414,
  },
  "placement": "bottom",
  "tooltipOrigin": Point {
    "x": 197,
    "y": 432,
  },
}
`;

exports[`Testing Computing Geometry correctly calculates bottom geometry 3`] = `
Object {
  "anchorPoint": Point {
    "x": 99,
    "y": 235,
  },
  "placement": "bottom",
  "tooltipOrigin": Point {
    "x": 0,
    "y": 243,
  },
}
`;

exports[`Testing Computing Geometry correctly calculates left geometry 1`] = `
Object {
  "anchorPoint": Point {
    "x": 24,
    "y": 382,
  },
  "placement": "left",
  "tooltipOrigin": Point {
    "x": -192,
    "y": 332,
  },
}
`;

exports[`Testing Computing Geometry correctly calculates left geometry 2`] = `
Object {
  "anchorPoint": Point {
    "x": 240,
    "y": 382,
  },
  "placement": "left",
  "tooltipOrigin": Point {
    "x": 83,
    "y": 282,
  },
}
`;

exports[`Testing Computing Geometry correctly calculates left geometry 3`] = `
Object {
  "anchorPoint": Point {
    "x": 24,
    "y": 135,
  },
  "placement": "left",
  "tooltipOrigin": Point {
    "x": -292,
    "y": 0,
  },
}
`;

exports[`Testing Computing Geometry correctly calculates right geometry 1`] = `
Object {
  "anchorPoint": Point {
    "x": 88,
    "y": 382,
  },
  "placement": "right",
  "tooltipOrigin": Point {
    "x": 104,
    "y": 332,
  },
}
`;

exports[`Testing Computing Geometry correctly calculates right geometry 2`] = `
Object {
  "anchorPoint": Point {
    "x": 304,
    "y": 382,
  },
  "placement": "right",
  "tooltipOrigin": Point {
    "x": 311,
    "y": 282,
  },
}
`;

exports[`Testing Computing Geometry correctly calculates right geometry 3`] = `
Object {
  "anchorPoint": Point {
    "x": 174,
    "y": 135,
  },
  "placement": "right",
  "tooltipOrigin": Point {
    "x": 190,
    "y": 0,
  },
}
`;

exports[`Testing Computing Geometry correctly calculates top geometry 1`] = `
Object {
  "anchorPoint": Point {
    "x": 56,
    "y": 350,
  },
  "placement": "top",
  "tooltipOrigin": Point {
    "x": 0,
    "y": 242,
  },
}
`;

exports[`Testing Computing Geometry correctly calculates top geometry 2`] = `
Object {
  "anchorPoint": Point {
    "x": 272,
    "y": 350,
  },
  "placement": "top",
  "tooltipOrigin": Point {
    "x": 197,
    "y": 132,
  },
}
`;

exports[`Testing Computing Geometry correctly calculates top geometry 3`] = `
Object {
  "anchorPoint": Point {
    "x": 99,
    "y": 35,
  },
  "placement": "top",
  "tooltipOrigin": Point {
    "x": 0,
    "y": -473,
  },
}
`;
